### PR TITLE
Separate `spec/ruby` to core and bundled gems

### PR DESCRIPTION
**Pull request by hsbt**
https://bugs.ruby-lang.org/issues/20187#note-8

I introduced `test-bundled-gems-spec` for mspec runner for bundled gems.

### CODE REVIEW

**Comment by hsbt** [comment]
Before:

```
3732 files, 33000 examples, 282647 expectations, 0 failures, 0 errors, 0 tagged
```

After:
```
3424 files, 32446 examples, 267562 expectations, 0 failures, 0 errors, 0 tagged
```

and

```
307 files, 554 examples, 6933 expectations, 0 failures, 0 errors, 0 tagged
```

**Comment by hsbt** [comment]
Further works:

* Enable other bundled gems like `matrix`, `prime` and others.
* Remove `.bundle` directory from `GEM_HOME` and others.

File: spec/default.mspec [hunk 1]
```diff
@@ -31,6 +31,16 @@ class MSpecScript
     ]
   end
 
+  # Disable to run for bundled gems in test-spec
+  set :bundled_gems, (File.readlines("#{srcdir}/gems/bundled_gems").map do |line|
+    next if /^\s*(?:#|$)/ =~ line
+    "#{srcdir}/spec/ruby/library/" + line.split.first
+  end.compact)
+  set :stdlibs, Dir.glob("#{srcdir}/spec/ruby/library/*")
+  set :library, get(:stdlibs).to_a - get(:bundled_gems).to_a
```
Comment (eregon) [inline spec/default.mspec:40]: FYI another way to do this is https://github.com/oracle/truffleruby/blob/93693825fb81c8ec875883bec587f11a49462a1f/spec/truffleruby.mspec#L77-L82.
But this is fine as well.

**Review by eregon — COMMENTED** [review]
This looks great, thank you!

**Pull request merged**